Core/Business/Addon/AddonManagerInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Business\Addon;
+
+interface AddonManagerInterface
+{
+    public function add(string $source);
+    public function remove(Addon $addon);
+
+    public function migrate(Addon $addon, string $version, string $source = null);
+    public function rollback(Addon $addon, string $version);
```

Comment:
Why is this both on the `AddonInterface` and on the `AddonManagerInterface`?

If we view the Addon / AddonManager relationship as an Entity / EntityManager relationship (the AddonManager's job is to "persist" addons on the store) , I'd even go as far as to say that we might not even need the AddonInterface. Think of Doctrine for instance: models are just any classes. No interface at all. Or Maybe `AddonInterface` could just hold the meta-data about the addon, such as type (theme, module, whatever else...), version, compatibility range, but no method that actually does something.